dts/bindings/bluetooth/nordic,nrf-dfe-ant.yaml

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+#
+# Note:
+#  All dfegpio[n]-pin properties have required property set to false. It
+#  depends on end user which Radio PSEL_GPIO[n] registers will be used to drive
+#  antenna switches.
+#  Nevertheless there is a requirement that at least two antennae are available.
+#  That means at least one dfegpio[n]-pin must be provided, to give Radio
+#  possibility to switch between these two antennae.
+
+description: |
+    Nordic DFE antennae matrix description
+
+compatible: "nordic,nrf-dfe-ant"
+
+include: [base.yaml]
+
+properties:
+    ant-num:
+      type: int
+      required: true
+      description: >
+        Number of available antennae.
+
+        Value zero means no antenna available. If provided value different than
+        zero, it must be greater or equal to two.
+
+    dfegpio0-pin:
+      type: int
+      required: false
+      description: Pin select for DFE pin 0
```

Comment:
We are trying to get away from `foo-pin` properties; see https://github.com/zephyrproject-rtos/zephyr/pull/30769. I would really rather not see any more of these added upstream; we're just going to have to deprecate and remove them.

@anangl , @pabigot : what would you say to modifying #30769 so we don't have to resolve the I2C question, merging that in, and rebasing this PR using a `dfe-gpios` phandle-array instead?

Reply:
Is there anything left to resolve on the I2C one?  0 is fine.

To switch to a phandle array: what would be used as the specifier for registers that aren't used?  There's no requirement that the enabled/configured GPIO settings be contiguous in the 8-register PSEL area.

Reply:
> Is there anything left to resolve on the I2C one? 0 is fine.

Yes, I believe so: https://github.com/zephyrproject-rtos/zephyr/pull/30769#issuecomment-747579873

I think this means we should support other flags in there, but TBH I'm not quite clear on the details and we might need a lot of discussion to get consensus, so I was thinking of decoupling that.



> To switch to a phandle array: what would be used as the specifier for registers that aren't used? There's no requirement that the enabled/configured GPIO settings be contiguous in the 8-register PSEL area.

2 options:

1. `df0-gpios`, `df1-gpios`, .... instead
2. an unspecified value in a phandle array can be given by setting the phandle to 0 and not giving any cells, like this: https://github.com/torvalds/linux/blob/0c6c887835b59c10602add88057c9c06f265effe/include/linux/of_gpio.h#L82

Option 2 doesn't work right now due to an edtlib bug. I'm looking into a fix.

Reply:
The register that this sets is `PSEL.DFEGPIO[n]`.  I think the name should reflect that so `dfegpio#` seems right to me.  `-pin` if this continues to use the ordinal; if we go with #30769 then `-gpios`.

Option 2 is probably the devicetree way, but I suspect it'd be pretty confusing

Reply:
> Option 2 doesn't work right now due to an edtlib bug. I'm looking into a fix.

https://github.com/zephyrproject-rtos/zephyr/pull/30847

Reply:
I've got three questions here: 
1. If I use -gpios instead of -pin, should I implement it as single property that is array of phandles or leave it as a separate properties for  each register as it is now ?
2. What about GPIO initialization sequence if the same GPIO is used for something else? Radio acquires those GPIOs only for IQ sampling during TX/RX. Besides that GPIOs are free to be used for something else. I do not say that it is a real world use case (someone would ask for that) but it is allowed by SOC spec. 
3. How can I translate port in -gpios property to a number? It is needed because value written to PSEL.DFEGPIO[n] have Nordic "encoding" of port and pin.